src/System.Private.ServiceModel/src/System/IdentityModel/Selectors/X509SecurityTokenProvider.cs

```diff
@@ -12,61 +12,39 @@ namespace System.IdentityModel.Selectors
     {
         private X509Certificate2 _certificate;
+        private bool _clone;
 
-        public X509SecurityTokenProvider(X509Certificate2 certificate)
+        public X509SecurityTokenProvider(X509Certificate2 certificate) : this(certificate, true) { }
+
+        internal X509SecurityTokenProvider(X509Certificate2 certificate, bool clone)
         {
             if (certificate == null)
             {
                 throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("certificate");
             }
 
-            _certificate = new X509Certificate2(certificate.Handle);
-        }
-
-        public X509SecurityTokenProvider(StoreLocation storeLocation, StoreName storeName, X509FindType findType, object findValue)
-        {
-            if (findValue == null)
+            _clone = clone;
+            if (_clone)
             {
-                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("findValue");
+                _certificate = new X509Certificate2(certificate.Handle);
             }
-
-            X509Store store = new X509Store(storeName, storeLocation);
-            X509Certificate2Collection certificates = null;
-            try
+            else
             {
-                store.Open(OpenFlags.ReadOnly);
-                certificates = store.Certificates.Find(findType, findValue, false);
-                if (certificates.Count < 1)
-                {
-                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new SecurityTokenException(SR.Format(SR.CannotFindCert, storeName, storeLocation, findType, findValue)));
-                }
-                if (certificates.Count > 1)
-                {
-                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new SecurityTokenException(SR.Format(SR.FoundMultipleCerts, storeName, storeLocation, findType, findValue)));
-                }
-
-                _certificate = new X509Certificate2(certificates[0].Handle);
+                _certificate = certificate;
             }
-            finally
-            {
-                System.ServiceModel.Security.SecurityUtils.ResetAllCertificates(certificates);
-                store.Dispose();
-            }
-        }
-
-        public X509Certificate2 Certificate
-        {
-            get { return _certificate; }
         }
 
         protected override async Task<SecurityToken> GetTokenCoreAsync(CancellationToken cancellationToken)
         {
-            return await Task.FromResult<SecurityToken>(new X509SecurityToken(_certificate));
+            return await Task.FromResult<SecurityToken>(new X509SecurityToken(_certificate, _clone, _clone));
         }
 
         public void Dispose()
         {
-            System.ServiceModel.Security.SecurityUtils.ResetCertificate(_certificate);
+            if (_clone)
+            {
+                System.ServiceModel.Security.SecurityUtils.ResetCertificate(_certificate);
+            }
```

Comment:
We don't on desktop. I'm hesitant to add extra checks when it hasn't been an issue. Also, at least at the moment, this is an internal only class so we are in full control of any usage.